File: problems/Design HashSet/myHashSet.cs
public class MyHashSet {

    /** Initialize your data structure here. */
    public MyHashSet() {

    }

    public void Add(int key) {
        if (!_store.Contains(key)) _store.Add(key);
    }

    public void Remove(int key) {
        if (_store.Contains(key)) _store.Remove(key);
    }

    /** Returns true if this set contains the specified element */
    public bool Contains(int key) {
        return _store.Contains(key);
    }

    private HashSet<int> _store = new HashSet<int>();
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.Add(key);
 * obj.Remove(key);
 * bool param_3 = obj.Contains(key);
 */

File: problems/Design HashSet/MyHashSet.java
class MyHashSet {

    public MyHashSet() {
        keyRange = 769;
        bucketArray = new Bucket[this.keyRange];

        for (int i = 0; i < this.keyRange; ++i)
            bucketArray[i] = new Bucket();
    }

    public void add(int key) {
        int bucketIndex = _hash(key);
        bucketArray[bucketIndex].insert(key);
    }

    public void remove(int key) {
        int bucketIndex = _hash(key);
        bucketArray[bucketIndex].delete(key);
    }

    public boolean contains(int key) {
        int bucketIndex = _hash(key);
        return bucketArray[bucketIndex].exists(key);
    }

    private Bucket[] bucketArray;
    private int keyRange;

    protected int _hash(int key) {
        return (key % keyRange);
    }

    class Bucket {
        private LinkedList<Integer> container;

        public Bucket() {
            container = new LinkedList<Integer>();
        }

        public void insert(Integer key) {
            int index = this.container.indexOf(key);
            if (index == -1) {
                container.addFirst(key);
            }
        }

        public void delete(Integer key) {
            container.remove(key);
        }

        public boolean exists(Integer key) {
            int index = container.indexOf(key);
            return (index != -1);
        }
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */
